### PR TITLE
GLOWS - Propagate bad-angle flags from L1B Histogram to L2 in GLOWS

### LINKED ISSUE #2723: ENH - Propagate bad-angle flags from L1B Histogram to L2 in GLOWS
Parent issue: #1276

Per Section 12.3.4 of the algorithm document, bad-angle flags from L1B must propagate to L2 using OR logic: if a given flag is `True` in **any** L1B histogram for the observation day, it is `True` in L2.

Currently `DailyLightcurve` sets `histogram_flag_array = np.zeros(...)` (placeholder) and there is a `# todo: bad angle filter` comment in `glows_l2.py:82`.

Implementation:
- Collect the `histogram_flag_array` from all good-time L1B blocks
- OR them together per bin to produce the L2 `histogram_flag_array`

Depends on #2721 (bad-angle flags must be non-zero in L1B first).

### PR DESCRIPTION
**Pull request by laspsandoval**
This pull request improves the handling and propagation of the `histogram_flag_array` in the GLOWS L2 data processing pipeline, ensuring that flags are correctly OR'd across all relevant L1B blocks per bin. It also updates and expands the test suite to verify this logic, including edge cases for empty datasets.

**Flag propagation logic:**

* Updated the `__post_init__` method in `glows_l2_data.py` to compute `histogram_flag_array` by OR'ing all flag values across epochs and flag rows for each bin, per Section 12.3.4, ensuring a flag is set in L2 if it is set in any L1B block.

**Test suite improvements:**

* Modified test fixtures in `test_glows_l2_data.py` to include the correct shape and initialization for `histogram_flag_array`, matching the updated logic. [[1]](diffhunk://#diff-85922de4b06d3b8b853330bbcaf5d45c1cac4405f6663481d8738eb2c19bd958R57-R76) [[2]](diffhunk://#diff-85922de4b06d3b8b853330bbcaf5d45c1cac4405f6663481d8738eb2c19bd958L111-R131)
* Added new tests to verify flag OR-propagation across epochs and flag rows, and to check that the output is all zeros when the input dataset is empty.

### CODE REVIEW

File: imap_processing/glows/l2/glows_l2_data.py [hunk 1]
```diff
@@ -90,7 +90,19 @@ def __post_init__(self, l1b_data: xr.Dataset) -> None:
...
 
-        self.histogram_flag_array = np.zeros(self.number_of_bins)
+        # OR histogram_flag_array across all good-time L1B blocks per bin.
+        # Per Section 12.3.4: a flag is True in L2 if it is True in any L1B block.
+        # flags shape: (n_epochs, 4, n_bins)
+        flags = l1b_data["histogram_flag_array"].data
+        if flags.size > 0:
+            # Flatten epochs and flag rows into one axis: (n_epochs * 4, n_bins)
+            flags_2d = flags.reshape(-1, self.number_of_bins)
+            # OR across all rows per bin: (n_bins,)
```
Comment (tech3371) [inline imap_processing/glows/l2/glows_l2_data.py]: The 'OR' at the beginning confused me. Now, I understand what you meant above and here. Can you make this minor change?

```suggestion
            # Apply binary 'OR' operation across all rows per bin: (n_bins,)
```

File: imap_processing/glows/l2/glows_l2_data.py [hunk 2]
```diff
@@ -90,7 +90,19 @@ def __post_init__(self, l1b_data: xr.Dataset) -> None:
         # TODO: Average this, or should they all be the same?
         self.spin_angle = np.average(l1b_data["imap_spin_angle_bin_cntr"].data, axis=0)
 
-        self.histogram_flag_array = np.zeros(self.number_of_bins)
+        # OR histogram_flag_array across all good-time L1B blocks per bin.
```
Comment (tech3371) [inline imap_processing/glows/l2/glows_l2_data.py]:
```suggestion
        # Apply 'OR' operation to histogram_flag_array across all good-time L1B blocks per bin.
```

**Review by tech3371 — APPROVED** [review]
LGTM
